File: offline-pet-mobile/ios/OfflinePet/UsageMonitorModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(UsageMonitorModule, NSObject)
RCT_EXTERN_METHOD(getForegroundPackage:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: offline-pet-mobile/ios/OfflinePet/UsageMonitorModule.swift
import Foundation

@objc(UsageMonitorModule)
class UsageMonitorModule: NSObject {
  @objc
  func getForegroundPackage(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    // iOS soft enforcement mode: no hard app blocking in MVP.
    resolve(nil)
  }
}
